src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiComponentIdentifier.cs

```diff
@@ -39,16 +44,33 @@ public ValueTask<ProjectComponents> GetComponentsAsync(IProject project, Cancell
                 components |= ProjectComponents.XamariniOS;
             }
 
-            var targetFrameworkNames = project.TargetFrameworks.Select(r => r.Name);
+            if (packageReferences.Any(x => x.Name.Equals(_xamarinFormsPackage, StringComparison.OrdinalIgnoreCase)) && project.TargetFrameworks.FirstOrDefault().Equals(TargetFrameworkMoniker.NetStandard20))
+            {
```

Comment:
do you only care about the First element being .NETStandard2.0? should this not be TargetFrameworks.Any(condition)?